src/colmap/geometry/sim3.cc

```diff
@@ -47,12 +48,23 @@ bool Sim3d::Estimate(const std::vector<Eigen::Vector3d>& src,
     return false;
   }
   CHECK_EQ(results.size(), 1);
-  scale = results[0].col(0).norm();
-  rotation = Eigen::Quaterniond(results[0].leftCols<3>() / scale).normalized();
-  translation = results[0].rightCols<1>();
+  *this = Sim3d::FromMatrix(results[0]);
   return true;
 }
 
+bool Sim3d::EstimateRobust(const std::vector<Eigen::Vector3d>& src,
```

Comment:
I would like to avoid taking a dependency on RANSAC in this basic sim3d class. I suggest to just inline this code in your new AlignReconstructions function instead.

Reply:
Done. Note that we already do have this dependency since `Sim3d -> colmap_estimators (SimilarityTransform) -> colmap_optim (LORANSAC)`. See https://github.com/colmap/colmap/pull/2173